### DISCUSSION TITLE
Can I pass kubeconfig options into Terratest?

### DISCUSSION THREAD

**Discussion by rhoboat** [gruntwork-io/knowledge-base#191] [Help]
Hi everyone, I'm testing a custom module with Terratest and wanted to know if there is a way to pass kubeconfig settings into the [Option struct](https://github.com/gruntwork-io/terratest/blob/7e38458132f4537cbeeccbd262fb3f2541ddb259/modules/terraform/options.go#L40) for the terraform module. Would I need to bring in the k8s provider?

**Comment by denis256** [comment]
Hi, depends on how module consumes kubeconfig settings - can be used `EnvVars` map to pass `KUBECONFIG` environment variable with the path to file with content, or can be used `Vars` map to pass configuration variables.

`EnvVars` passing
https://github.com/gruntwork-io/terratest/blob/master/modules/terraform/apply_test.go#L196

`Vars` passing:
https://github.com/gruntwork-io/terratest/blob/master/test/terraform_aws_ecs_example_test.go#L33

https://github.com/gruntwork-io/terratest/blob/master/examples/terraform-aws-ecs-example/variables.tf#L22

**Reply by Abhi94N** [reply]
hi @denis256. I tried to pass the **KUBECONFIG** envar as shown below
```
		EnvVars: map[string]string{
			"KUBECONFIG": "/root/.kube/config",
		}
```

and I am getting the following error

` Error: Kubernetes cluster unreachable: invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable`

I'm not sure what value I'm supposed to pass here. but found that the setting has been deprecated. https://github.com/kubernetes/kubernetes/pull/96535

**Reply by denis256** [reply]
Hi,
will be good to know how K8S cluster is accessed, if it is used `terraform-aws-modules/eks/aws`, try setting `KUBE_CONFIG_PATH`
